src/rt/runtime.rs

```diff
@@ -258,19 +260,19 @@ impl Machine {
                 continue;
             }
 
-            let mut sched = rt.sched.lock().unwrap();
-
+            // If another thread is already blocked on the reactor, there is no point in keeping
+            // the current thread around since there is too little work to do.
             if sched.polling {
-                thread::sleep(Duration::from_micros(10));
-                continue;
+                break;
             }
```

Comment:
I'm not sure, terminating the thread is a good idea, what will happen when the load is increasing? no new thread will be created

simple sleep like this will end up with single thread
```rust
        task::spawn(async {
            loop {
                task::sleep(Duration::from_secs(1)).await;
                io::stdout().write_all(b"Hello World\n").await.unwrap();
            }
        })
        .await;
```

Reply:
I want to bring back auto spawning thread here https://github.com/async-rs/async-std/issues/740 , but it's just as a fallback if the users forget to use `spawn_blocking` (or especially when they don't own full control of the code, like using a library)

Reply:
We can not go back to the sleep & continue, it regresses performance even further, so we need to find a better solution.

Reply:
The correct fix as far as I understand is to bring back the dynamic restarting of machines: https://github.com/async-rs/async-std/pull/748